pkg/console/controllers/oauthclients/oauthclients.go

```diff
@@ -163,47 +173,42 @@ func (c *oauthClientsController) sync(ctx context.Context, controllerContext fac
 		waitCtx, cancel := context.WithTimeout(ctx, 10*time.Second)
 		defer cancel()
 		if !cache.WaitForCacheSync(waitCtx.Done(), c.oauthClientSwitchedInformer.Informer().HasSynced) {
-			syncErr = fmt.Errorf("timed out waiting for OAuthClients cache sync")
-			break
+			return statusHandler.FlushAndReturn(fmt.Errorf("timed out waiting for OAuthClients cache sync"))
 		}
 
 		clientSecret, secErr := c.syncSecret(ctx, operatorConfig, controllerContext.Recorder())
 		statusHandler.AddConditions(status.HandleProgressingOrDegraded("OAuthClientSecretSync", "FailedApply", secErr))
 		if secErr != nil {
-			syncErr = secErr
-			break
+			return statusHandler.FlushAndReturn(secErr)
 		}
 
 		oauthErrReason, oauthErr := c.syncOAuthClient(ctx, clientSecret, consoleURL.String())
 		statusHandler.AddConditions(status.HandleProgressingOrDegraded("OAuthClientSync", oauthErrReason, oauthErr))
 		if oauthErr != nil {
-			syncErr = oauthErr
-			break
+			return statusHandler.FlushAndReturn(oauthErr)
 		}
 
 	case configv1.AuthenticationTypeOIDC:
 		syncErr = c.syncAuthTypeOIDC(ctx, controllerContext, statusHandler, operatorConfig, authnConfig)
 		if syncErr != nil {
-			break
+			return statusHandler.FlushAndReturn(syncErr)
```

Comment:
Because this combines multiple fields, this early return breaks the AuthStatusHandler. do you want a separate controller?